### ISSUE TITLE
Investigate `forkAff`

### ISSUE THREAD

**Issue by cprussin** [cprussin/purescript-httpure#111]
I believe that we should be using `forkAff` when running the router.  We should investigate implications of doing that.

**Comment by akheron** [comment]
What would you like to achieve by using `forkAff`?

**Comment by cprussin** [comment]
Because `forkAff` causes the `Aff` to be non-blocking, it may help improve concurrency.  For instance, if a particular route requires reading a file or going to a database, I think without using `forkAff` we'll be blocking the main thread rather than concurrently handling requests.

**Comment by akheron** [comment]
In the context of `forkAff`, being non-blocking means that the code lexically following `forkAff` continues to run even if the `Aff` computation didn't finish yet (didn't call the success or error callback yet).

Reading from a database is in a different "layer", in the sense that NodeJS will continue serving new requests concurrently even if one request handler is waiting for database (or any other I/O). If the database connection would be blocking in the sense the term is used in NodeJS, then `forkAff` doesn't help. The thread is blocked and cannot do anything else. Blocking the thread is highly discouraged and libraries rarely do it.

tl;dr There's no need to use `forkAff` in `HTTPure.serve` to support multiple concurrent requests.

**Comment by cprussin** [comment]
Ah, ok, that's a good explanation, thanks.  I was worried that without using it we could be blocking further calls to `Server.handleRequest`, but I guess that given the nature of Node under the hood that wouldn't really make sense.  Thanks for taking the time to write it up!
